libcontainerd/remote_linux.go

```diff
@@ -388,7 +391,9 @@ func (r *remote) runContainerdDaemon() error {
 		return err
 	}
 
-	go cmd.Wait() // Reap our child when needed
+	go func() {
+		r.daemonWaitCh <- cmd.Wait()
```

Comment:
I think there will be goroutine leak in case of `!utils.IsProcessAlive(r.daemonPid)`. Make channel buffered or if you're not going to use error, better to do it `chan struct{}` and just close after wait.